runtime/resolver.go

```diff
@@ -40,6 +40,8 @@ type Resolver interface {
 type ResolverResult struct {
 	// Data is a JSON encoded array of objects.
 	Data []byte
+	// Rows is deserialized array of objects. If Rows is set Data will be unset and vice versa.
+	Rows [][]any
```

Comment:
Yes. That makes sense. 
From an implementation POV, does this mean that `MarshalJSON` will use `NextRow` and produce results ?

Reply:
I would probably add a `sqlResolverResult` type in the `resolvers` package that gets a `database/sql.Rows` object and implements the various methods. For the `MarshalJSON` option, it can keep the implementation currently used for `Data`.

Reply:
One problem with streaming results is that we can't cache the results.